zenoh/src/publication.rs

```diff
@@ -117,6 +121,22 @@ impl PutBuilder<'_, '_> {
         self.kind = kind;
         self
     }
+
+    #[zenoh_macros::unstable]
+    pub fn attachment(&self) -> Option<&Attachment> {
```

Comment:
Looking at this more closely. What's the interest of `attachment` and `attachment_mut` in the `PutBuilder`? Shouldn't `with_attachment` be enough? It's a builder, so I would expect to be additive for the attachment...

Reply:
Agreed for builders